sympy/physics/mechanics/body.py

```diff
@@ -0,0 +1,165 @@
+from sympy import Symbol, symbols
+from sympy.physics.mechanics import (RigidBody, Particle, ReferenceFrame,
+                                     inertia)
+from sympy.physics.vector import Point, Vector
+
+__all__ = ['Body']
+
+
+class Body(RigidBody, Particle):
+    """
+    Body is a common representation of RigidBody or a Particle.
+
+    A Body represents either a rigid body or particle in classical mechanics.
+    Bodies have a body-fixed reference frame, a mass, a mass center and
+    possibly a body-fixed inertia.
+
+    Parameters
+    ----------
+    name: String
+        Defines the name of the body. It is used as the base for defining body
+        specific properties.
+    masscenter : Point, optional
+        The point which represents the center of mass of the rigid body.
+    frame : ReferenceFrame (optional)
+        The ReferenceFrame in which the rigid body is fixed.
+    mass : Sympifyable, optional
+        The body's mass.
+    body_inertia : Dyadic (instance of inertia)
+        The body's inertia about center of mass.
+
+    Examples
+    --------
+    1. Default behaviour. It creates a RigidBody after defining mass,
+     mass center, frame and inertia.
+
+    >>> from sympy.physics.mechanics import Body
+    >>> body = Body('name_of_body')
+
+    2. Passing attributes of Rigidbody. All the arguments needed to create a
+     RigidBody can be passed while creating a Body too.
+
+    >>> from sympy import Symbol
+    >>> from sympy.physics.mechanics import ReferenceFrame, Point, inertia
+    >>> from sympy.physics.mechanics import Body
+    >>> mass = Symbol('mass')
+    >>> masscenter = Point('masscenter')
+    >>> frame = ReferenceFrame('frame')
+    >>> ixx = Symbol('ixx')
+    >>> body_inertia = inertia(frame, ixx, 0, 0)
+    >>> body = Body('name_of_body', masscenter, mass, frame, body_inertia)
+
+    3. Creating a Particle. If masscenter and mass are passed, and inertia is
+     not then a Particle is created.
+
+    >>> from sympy import Symbol
+    >>> from sympy.physics.vector import Point
+    >>> from sympy.physics.mechanics import Body
+    >>> mass = Symbol('mass')
+    >>> masscenter = Point('masscenter')
+    >>> body = Body('name_of_body', masscenter, mass)
+
+    Similarly, A frame can also be passed while creating a Particle.
+
+    """
+    def __init__(self, name, masscenter=None, mass=None, frame=None,
+                 body_inertia=None):
+
+        self.name = name
+        self.loads = []
+
+        if masscenter is None:
+            _masscenter = Point(name + '_masscenter')
+        else:
+            _masscenter = masscenter
```

Comment:
Is there a particular reason why mass center is being used, rather than the more common term center/centre of mass?